### PR TITLE
:recycle: Refactors away complexity

### PR DESCRIPTION
**Pull request by ekwoka**
So I think going for a simpler approach should solve most issues.

This I think solves most of your auto complete issue.

It also improves some other behaviors like reformatting numbers as you hit backspace.

All with less code and more performance.

🥂 

It does change one behavior of mask which might need some more consideration as to whether it should change or not.

### CODE REVIEW

File: packages/mask/src/index.js [hunk 1]
```diff
@@ -109,137 +99,52 @@ export function restoreCursorPosition(el, template, callback) {
...
 
     el.setSelectionRange(newPosition, newPosition)
 }
 
+let regexes = {
+    '9': /[0-9]/,
+    'a': /[a-zA-Z]/,
+    '*': /[a-zA-Z0-9]/,
+};
 export function stripDown(template, input) {
```
Comment (robertmarney) [inline packages/mask/src/index.js:114]: Question: Is this required any more? I do not believe it is referenced anywhere other than module.js in order to consume in jest testing?

**Review by robertmarney — APPROVED** [review]
Hi @ekwoka - Apologies for the delay, I have been battling a stomach flu and have been bed-ridden for a few days. Thank you for taking on the refactor!

I am going to merge this into a new working branch on this repository then update the jest spec and look to add a few more cypress tests to validate the initial complaint.

You mentioned a change in behavior, what specifically were you referring too?

File: packages/mask/src/index.js (hunk 1 above)
Comment (ekwoka) [reply packages/mask/src/index.js:114]: I was just thinking that, in the interest of good UX, having it also exported could be useful for people that want to strip out the input values themselves.

Like a money one where they just want the numbers as raw ints or something. Heck if I know. Seems valuable in the general sense, not the specific.

Comment (robertmarney) [reply packages/mask/src/index.js:114]: Fair enough I can re-introduce, I guess my concern is that we are no longer directly consuming the output within the new `formatInput` implementation that its not really clear what the output represents and its functionality could deviate from future change of `formatInput`.

Comment (ekwoka) [reply packages/mask/src/index.js:114]: I'm not sure what you mean. The code is consuming the output from formatInput.

formatInput is a pure function.

**Pull request merged**